sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -15,15 +15,16 @@ namespace Azure.Core
     internal abstract class OperationInternalBase
     {
         private readonly ClientDiagnostics _diagnostics;
-        private readonly string _updateStatusScopeName;
         private readonly IReadOnlyDictionary<string, string>? _scopeAttributes;
         private readonly DelayStrategy? _fallbackStrategy;
         private readonly AsyncLockWithValue<Response> _responseLock;
 
+        protected readonly string _operationTypeName;
+
         protected OperationInternalBase(Response rawResponse)
         {
             _diagnostics = new ClientDiagnostics(ClientOptions.Default);
-            _updateStatusScopeName = string.Empty;
+            _operationTypeName = string.Empty;
```

Comment:
We would now open a scope called ".UpdateStatus" or ".WaitForCompletionResponse" if someone used this constructor.  We shouldn't rely on the fact that there is only one caller of this constructor (right now) to ensure that we don't get this behavior -- it would be easy for someone to add a usage of this constructor that _does_ open a scope.

Reply:
This is never going to happen. If anyone calls this constructor, he has to pass in a rawResponse, which will initiliza a `_responseLock` by storing its value. With this value, we will stop before we create the scopes. We will return [here](https://github.com/Azure/azure-sdk-for-net/blob/1660eec291fea10fb590eee75eacbdeeb6f20675/sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs#L188).

Reply:
This constructor is for completed operation only. As @pshao25 has correctly pointed out, no scope should be opened for completed operation. If that happens, then whole type is broken

Reply:
I see, thanks for helping me understand!  Per offline conversation, I understand now that we can guarantee that this constructor is only ever called with a Response from a completed operation because:

1. It can only be called by subclasses, because it is protected.
2. Only OperationInternal and OperationInternal\<T\> subclass OperationInternalBase, and so are the only callers of this constructor.
3. The only callers of the protected constructor are on static Succeeded() and Failed() factory methods, that are only ever called after the operation has completed.

Thank you!

Would it make sense to set _operationTypeName to null, then, to indicate that we expect it will never be used in a concatenated string?

Reply:
I see, thanks for helping me understand!  Per offline conversation, I understand now that we can guarantee that this constructor is only ever called with a Response from a completed operation because:

1. It can only be called by subclasses, because it is protected.
2. Only OperationInternal and OperationInternal\<T\> subclass OperationInternalBase, and so are the only callers of this constructor.
3. The only callers of the protected constructor are on static Succeeded() and Failed() factory methods, that are only ever called after the operation has completed.

Thank you!

Would it make sense to set _operationTypeName to null, then, to indicate that we expect it will never be used in a concatenated string?

Reply:
Agree. Will update after we have a conclusion for the perf issue of https://github.com/Azure/azure-sdk-for-net/pull/29033#discussion_r891517743

Reply:
> Would it make sense to set _operationTypeName to null

It would require to add null checks in various places where field is used. So I'd prefer it to be `string.Empty`. Also, I'd like to bring back `_updateStatusScopeName` to avoid creation of garbage strings.